File: rb/features.rb
# AntiPhishingDetection SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module AntiPhishingDetectionFeatures
  def self.make_feature(name)
    case name
    when "base"
      AntiPhishingDetectionBaseFeature.new
    when "ratelimit"
      AntiPhishingDetectionRatelimitFeature.new
    when "retry"
      AntiPhishingDetectionRetryFeature.new
    when "test"
      AntiPhishingDetectionTestFeature.new
    when "timeout"
      AntiPhishingDetectionTimeoutFeature.new
    else
      AntiPhishingDetectionBaseFeature.new
    end
  end
end

File: py/antiphishingdetection_sdk/features.py
# AntiPhishingDetection SDK feature factory

from antiphishingdetection_sdk.feature.base_feature import AntiPhishingDetectionBaseFeature
from antiphishingdetection_sdk.feature.test_feature import AntiPhishingDetectionTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AntiPhishingDetectionBaseFeature(),
        "test": lambda: AntiPhishingDetectionTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
